ds3-autogen-python/src/main/java/com/spectralogic/ds3autogen/python/PythonCodeGenerator.java

```diff
@@ -15,15 +15,125 @@
+import com.spectralogic.ds3autogen.python.model.commands.CommandSet;
+import com.spectralogic.ds3autogen.python.model.request.BaseRequest;
+import com.spectralogic.ds3autogen.utils.Helper;
+import com.spectralogic.ds3autogen.utils.collections.GuavaCollectors;
+import freemarker.template.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
 import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.spectralogic.ds3autogen.utils.ConverterUtil.isEmpty;
+import static com.spectralogic.ds3autogen.utils.ConverterUtil.removeUnusedTypes;
 
 public class PythonCodeGenerator implements CodeGenerator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PythonCodeGenerator.class);
+    private static final Path BASE_PROJECT_PATH = Paths.get("ds3");
+
+    private final Configuration config = new Configuration(Configuration.VERSION_2_3_23);
+
+    private Ds3ApiSpec spec;
+    private FileUtils fileUtils;
+    private Path destDir;
+
+    public PythonCodeGenerator() throws TemplateModelException {
+        config.setDefaultEncoding("UTF-8");
+        config.setTemplateExceptionHandler(TemplateExceptionHandler.RETHROW_HANDLER);
+        config.setClassForTemplateLoading(PythonCodeGenerator.class, "/tmpls");
+        config.setSharedVariable("pythonHelper", PythonHelper.getInstance());
+        config.setSharedVariable("helper", Helper.getInstance());
+    }
+
     @Override
     public void generate(final Ds3ApiSpec spec, final FileUtils fileUtils, final Path destDir) {
+        this.spec = spec;
+        this.fileUtils = fileUtils;
+        this.destDir = destDir;
+
+        try {
+            final ImmutableList<Ds3Request> requests = spec.getRequests();
+            final ImmutableMap<String, Ds3Type> typeMap = removeUnusedTypes(
+                    spec.getTypes(),
+                    spec.getRequests());
+
+            generateCommands(requests);
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Generates the python code for commands
+     */
+    private void generateCommands(final ImmutableList<Ds3Request> ds3Requests) throws IOException, TemplateException {
+        if (isEmpty(ds3Requests)) {
+            LOG.info("There are no requests to generate");
+            return;
+        }
+
+        final ImmutableList<BaseRequest> baseRequests = toRequestModelList(ds3Requests);
+        final CommandSet commandSet = new CommandSet(baseRequests);
+
+        final Template tmpl = config.getTemplate("python/commands/all_commands.ftl");
+        final Path path = toBaseProjectPath("ds3.py");
+
+        LOG.info("Getting OutputStream for file: " + path.toString());
+
+        try (final OutputStream outStream = fileUtils.getOutputFile(path);
+             final Writer writer = new OutputStreamWriter(outStream)) {
```

Comment:
While probably not required, writer should be closed after use, so it's declaration should be moved into the try-with-resource block.